File: easy/count-vowels/python/count_vowels.py
def count_vowels(s: str) -> int:
    """Count vowels in a given string."""
    vowels = set("aeiouAEIOU")
    return sum(1 for ch in s if ch in vowels)


if __name__ == "__main__":
    tests = ["hello world", "AEIOU", "xyz", "", "ChatGPT rocks!"]
    for t in tests:
        print(f"'{t}' → {count_vowels(t)}")

File: easy/count-vowels/golang/count_vowels.go
package countvowels

import "strings"

// CountVowels returns the number of vowels in a string
func CountVowels(s string) int {
	count := 0
	vowels := "aeiouAEIOU"
	for _, char := range s {
		if strings.ContainsRune(vowels, char) {
			count++
		}
	}
	return count
}
